### PR TITLE
[ticket/13468] Change add_log() calls with $phpbb_log->add()

### PR DESCRIPTION
**Pull request by MGaetan89**
https://tracker.phpbb.com/browse/PHPBB3-13468

PHPBB3-13468

### CODE REVIEW

**Comment by MGaetan89** [comment]
!set WIP

File: phpBB/develop/create_search_index.php [hunk 1]
```diff
@@ -132,7 +132,7 @@
 
 $search->tidy();
 
-add_log('admin', 'LOG_SEARCH_INDEX_CREATED', $search_name);
+$phpbb_log->add('admin', $user->data['user_id'], $user->ip, 'LOG_SEARCH_INDEX_CREATED', time(), array($search_name));
```
Comment (Nicofuma) [inline phpBB/develop/create_search_index.php]: could be better to use false instead of time() here... I don't really know...

**Comment by Nicofuma** [comment]
Seems to be good for now, don't forgot the other calls :)

File: phpBB/develop/create_search_index.php (hunk 1 above)
Comment (MGaetan89) [inline phpBB/develop/create_search_index.php]: I didn't think about that. But I think `0` would be a better default value for `log_interface::add()`, as the documentation says it should be an int.

Comment (nickvergessen) [inline phpBB/develop/create_search_index.php]: although 0 is a valid timestamp, so maybe null instead?
Or we just fix the interface

Comment (nickvergessen) [inline phpBB/develop/create_search_index.php]:
```php
		if ($log_time == false)
		{
			$log_time = time();
		}
```
Should be false, also there should be 3 equals ;)
So mind fixing the interface and the implementation?

Comment (MGaetan89) [inline phpBB/develop/create_search_index.php]: So `null` by default and 3 equals?

Comment (nickvergessen) [inline phpBB/develop/create_search_index.php]: false by default and 3 equals

Comment (MGaetan89) [inline phpBB/develop/create_search_index.php]: Change is done.

**Comment by MGaetan89** [comment]
Some tests failed because I used `$user->data['user_id']` instead of `ANONYMOUS` in some places.
Instead of checking the value of `$user->data['user_id']` everywhere, I added the method `$user->get_user_id()`.
If that solution is ok, I will keep the last commit, and use this method for the other `add_log()` that needs to be changed. Otherwise I will keep the checks where it was previously used.

**Comment by nickvergessen** [comment]
I'd fix that in https://github.com/phpbb/phpbb/blob/develop/phpBB/phpbb/log/log.php#L232
`'user_id'	=> ($user_id) ? (int) $user_id : ANONYMOUS,`

**Comment by MGaetan89** [comment]
So much easier... Thanks, I'll update the code.

**Comment by MGaetan89** [comment]
I removed the `$user->get_user_id()` method. I will try to finish this PR tomorrow.

**Comment by MGaetan89** [comment]
Is it valid to use a default value instead of a type?

**Comment by MGaetan89** [comment]
I updated the remaining `add_log()` calls and move the function to `functions_compatibility.php`.
Next step is fixing the tests and then the PR is good to do.

**Comment by MGaetan89** [comment]
!unset WIP

**Comment by MGaetan89** [comment]
Test are all passing now. PR is ready for review/merge.

File: phpBB/develop/create_search_index.php (hunk 1 above)
Comment (Nicofuma) [inline phpBB/develop/create_search_index.php]: You still need to use false instead of time() here and in a few other places

Comment (MGaetan89) [inline phpBB/develop/create_search_index.php]: But if I set `$log_time` to `false`, `time()` will be use. So it's the same, no?

**Comment by nickvergessen** [comment]
needs a rebase (because of your type hint PR)

**Comment by MGaetan89** [comment]
Done. Tests are running.
What is the final decision about using `false` vs. `time()` for the `$log_time` parameter?

**Comment by Nicofuma** [comment]
I'll say `false` @nickvergessen ?

**Comment by nickvergessen** [comment]
`false` and triple equals please as stated above :)

**Comment by MGaetan89** [comment]
`===` is already there. I will change `time()` with `false` and update the PR :wink:

**Comment by MGaetan89** [comment]
Updated and tests are passing.

**Comment by Nicofuma** [comment]
Can you rebase again please?

**Comment by MGaetan89** [comment]
Rebased. Tests are running.

**Comment by MGaetan89** [comment]
Tests are passing :smile:

**Comment by MGaetan89** [comment]
Rebased on `develop`.

**Comment by Nicofuma** [comment]
Please, fix some comments by the way (add_log() will disappear one day ^^)

functions_user.php:line 317: `// Because these actions only fill the log unneccessarily we disable it`
phpbb\log\log.php:line 30: `* added when add() is called.`
phpbb\log\log_interface.php:line 35: `* page call. When add() is called and the type is disabled,`
test\dbal\migrator_tool_module_test.php: line 30: `// Disable the logs`

**Comment by MGaetan89** [comment]
Good catch.
Changes are done.

**Comment by Nicofuma** [comment]
Well, should be good now :) I let it open a few another hours and will merge it tomorrow :)

**Pull request merged**